### PR TITLE
Feat/prefetch customui android

### PR DESCRIPTION
**Pull request by KausthubhRzr**
Implemented Turbo Android Prefetch UI Dart Wrapper
- Prefetch UI Implementation
- Set UPI Pin for Prefetch

### CODE REVIEW

File: android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java [hunk 1]
```diff
@@ -621,18 +652,122 @@ public void onSuccess(@NonNull List<UpiAccount> upiAccounts) {
...
+                        public void onError(@NonNull Error error) {
+                            pendingResult.error(error.getErrorCode(), error.getErrorDescription(), toJsonString(error));
+                        }
+                    });
+        }
+
+    }
+
+
+//    func setPrefetchUPIPinWithUI(bankAccountStr: String , result: @escaping FlutterResult, eventSink: @escaping FlutterEventSink){
```
Comment (ramprasadAnand) [inline android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java:741]: Remove if not required @KausthubhRzr

File: android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java [hunk 2]
```diff
@@ -621,18 +652,122 @@ public void onSuccess(@NonNull List<UpiAccount> upiAccounts) {
...
+                                    Log.e("Exception Occurred", "Pin-set account parsing Exception");
+                                }
+                            }
+
+
+                            Map<String, Object> finalDict = new HashMap<>();
+                            finalDict.put("accountsWithPinNotSet", pinNotSetArr);
+                            finalDict.put("accountsWithPinSet", pinSetArr);
+
+                            String finalDictStr = new Gson().toJson(finalDict);
```
Comment (ramprasadAnand) [inline android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java:690]: @KausthubhRzr can you please rename `finalDictStr` variable as this data is not being sent as string to flutter wrapper.

**Review by ramprasadAnand — COMMENTED** [review]
Added few comments

File: android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java (hunk 2 above)
Comment (KausthubhRzr) [reply android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java:690]: @ramprasadAnand in the above code snippet,the HashMap is getting converted to String. Hence I have added the name in the manner

File: android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java (hunk 1 above)
Comment (KausthubhRzr) [reply android/src/main/java/com/razorpay/flutter_customui/RazorpayDelegate.java]: @ramprasadAnand Done

**Review by ramprasadAnand — APPROVED** [review]
LGTM

File: android/build.gradle [hunk 3]
```diff
@@ -54,7 +54,7 @@ android {
         implementation 'com.squareup.retrofit2:converter-gson:2.9.0'
         implementation 'com.squareup.okhttp3:logging-interceptor:4.10.0'
         //TODO: Add Library folder path in implementation's dir
-        //compileOnly fileTree(include: ['*.aar'], dir: '')
+        compileOnly fileTree(include: ['*.aar'], dir: '/Users/kausthubh.adhikari/Documents/GitHub/UAT-v1.2.5')
```
Comment (vivekshindhe) [inline android/build.gradle:57]: @KausthubhRzr can we revert this?

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #Wed Aug 21 14:55:09 IST 2024
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.9-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip
```
Comment (vivekshindhe) [inline android/gradle/wrapper/gradle-wrapper.properties:3]: @KausthubhRzr can you please specify the reason for the downgrade?

Comment (KausthubhRzr) [reply android/gradle/wrapper/gradle-wrapper.properties:3]: @vivekshindhe the base branch for my branch is [feat/preftech-customui-ios](https://github.com/razorpay/razorpay-flutter-customui/tree/feat/preftech-customui-ios) in this branch the gradle version was updated, but the master is on [6.7-all.zip](https://github.com/razorpay/razorpay-flutter-customui/blob/turbo-flutter-release/android/gradle/wrapper/gradle-wrapper.properties). Will be picking this up separately.

**Pull request merged**